app/src/main/java/com/kickstarter/ui/activities/ThreadActivity.kt

```diff
@@ -167,6 +169,48 @@ class ThreadActivity :
                     )
                 )
             }
+
+        viewModel.outputs.hasPendingComments()
+            .compose(bindToLifecycle())
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe {
+                if (it) {
```

Comment:
instead of an if/else block, can we have use the `filter` operator to only emit if true call the handleBackAction? or do we have to explicitly call `viewmodel.input.backPressed()` in the else block? either way, this can be shorted to  one line `if (it) handleBackAction() else viewmodel.inputs.backpressed()`

Reply:
we want to user press back and had pending comment to show dialog to confirm back
so we have to intercept back action and check if there is pending comments then show dialog  if not back

Reply:
changed to if (it) handleBackAction() else viewmodel.inputs.backpressed()